efi/preinstall/result.go

```diff
@@ -0,0 +1,550 @@
+	// This check may not run if the NoSecureBootPolicyProfileSupport flag is set.
+	//
+	// Note that this flag is not persisted when serializing the results.
+	WeakSecureBootAlgorithmsDetected
+
+	// PreOSVerificationUsingDigestDetected indicates that pre-OS components were verified by the
+	// use of a digest hardcoded in the authorized signature database as opposed to a X.509 certificate.
+	// Support for this has to be opted into with the PermitPreOSVerificationUsingDigests flag to
+	// RunChecks, as it implies that db has to change with each update to certain firmware components.
+	// This check may not run if the NoSecureBootPolicyProfileSupport flag is set.
+	//
+	// Note that this flag is not persisted when serializing the results.
+	PreOSVerificationUsingDigestsDetected
+)
+
+var checkResultFlagToIDStringMap = map[CheckResultFlags]string{
+	NoPlatformFirmwareProfileSupport:     "no-platform-firmware-profile-support",
+	NoPlatformConfigProfileSupport:       "no-platform-config-profile-support",
+	NoDriversAndAppsProfileSupport:       "no-drivers-and-apps-profile-support",
+	NoDriversAndAppsConfigProfileSupport: "no-drivers-and-apps-config-profile-support",
+	NoBootManagerCodeProfileSupport:      "no-boot-manager-code-profile-support",
+	NoBootManagerConfigProfileSupport:    "no-boot-manager-config-profile-support",
+	NoSecureBootPolicyProfileSupport:     "no-secure-boot-policy-profile-support",
+	DiscreteTPMDetected:                  "discrete-tpm-detected",
+	StartupLocalityNotProtected:          "startup-locality-not-protected",
+}
+
+var checkNonPersistentResultFlagToIDStringMap = map[CheckResultFlags]string{
+	VARDriversPresent:                     "var-drivers-present",
+	SysPrepApplicationsPresent:            "sysprep-apps-present",
+	AbsoluteComputraceActive:              "absolute-active",
+	NotAllBootManagerCodeDigestsVerified:  "not-all-boot-manager-code-digests-verified",
+	RunningInVirtualMachine:               "running-in-vm",
+	WeakSecureBootAlgorithmsDetected:      "weak-secure-boot-algs-detected",
+	PreOSVerificationUsingDigestsDetected: "pre-os-verification-using-digests-detected",
+}
+
+var checkResultFlagFromIDStringMap = map[string]CheckResultFlags{
+	"no-platform-firmware-profile-support":       NoPlatformFirmwareProfileSupport,
+	"no-platform-config-profile-support":         NoPlatformConfigProfileSupport,
+	"no-drivers-and-apps-profile-support":        NoDriversAndAppsProfileSupport,
+	"no-drivers-and-apps-config-profile-support": NoDriversAndAppsConfigProfileSupport,
+	"no-boot-manager-code-profile-support":       NoBootManagerCodeProfileSupport,
+	"no-boot-manager-config-profile-support":     NoBootManagerConfigProfileSupport,
+	"no-secure-boot-policy-profile-support":      NoSecureBootPolicyProfileSupport,
+	"discrete-tpm-detected":                      DiscreteTPMDetected,
+	"startup-locality-not-protected":             StartupLocalityNotProtected,
+}
+
+type x509CertificateIdJSON struct {
```

Comment:
nitpick: I would split out the x509*Id* types to maybe their own x509id.go file

Reply:
I've moved these types into their own file.